drivers/platform/aducm3029/rtc.c

```
@@ -0,0 +1,186 @@
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/************************* Include Files **************************************/
+/******************************************************************************/
+
+#include <stdlib.h>
+#include "error.h"
+#include "rtc.h"
+#include "rtc_extra.h"
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+
+/**
+ * @brief Initialize the RTC peripheral.
+ * @param device - The RTC descriptor.
+ * @param init_param - The structure that contains the RTC initialization.
+ * @return SUCCESS in case of success, FAILURE otherwise.
+ */
+int32_t rtc_init(struct rtc_dev **device, struct rtc_init_param *init_param)
+{
+	int32_t ret;
+	struct rtc_dev *dev;
+	struct aducm_rtc_dev *adev;
+	struct aducm_rtc_init_param *ainit = init_param->extra;
+
+	dev = (struct rtc_dev *)calloc(1, sizeof(*dev));
+	if (!dev)
+		return FAILURE;
```

I believe the goto statements are shifted.
Instead of returning failure, you should use goto error_dev( same for the other).